lld/wasm/LTO.cpp

```diff
@@ -40,6 +41,7 @@ using namespace llvm;
 namespace lld::wasm {
 static std::unique_ptr<lto::LTO> createLTO() {
   lto::Config c;
+  c.EmbedCmdArgs = commonContext().cmdArgs;
```

Comment:
It looks like `commonContext()` is not actually used elsewhere in lld, and this should probably just be `context()`?

Reply:
Thanks for the comment! I don't think it makes a difference, except that `context()` needs a `T` (which derives from `CommonLinkerContext` regardless) and this would result in more typing to get the exact same instance.

Reply:
How about adding a helper like the existing ones `CommonLinkerContext.h`:

```
inline llvm::StringSaver &saver() { return context().saver; }
```

This means that the current linker code just calls `saver()` whenever it needs it.

Reply:
(I'm not sure why `context()` needs to take a template param, and it looks like its optional.. none of the current uses seem to specify one).

Reply:
For most drivers (except ELF) I can access the specialized context directly. But when creating the LTO config this isn’t the case. I can add a helper function, but it only complicates the code in my opinion. Tomorrow I will take another look to see if I missed something.

Reply:
I think it might be the other way around.  It looks like only COFF currently supported specialized contexts.  The one I work on (Wasm) currently doesn't, for example.

Reply:
I changed all the `commonContext()` to `context()` (or the `ctx` variable directly where possible). Feel free to reopen the discussion if you think it needs another revision.